src/Ninja/G1CP/Content/Tests/test203.d

```diff
@@ -0,0 +1,107 @@
+
+    // Check if dialog exists
+    var int infoId; infoId = MEM_GetSymbolIndex(infoSymbName);
+    if (infoId == -1) {
+        G1CP_TestsuiteErrorDetail(ConcatStrings(ConcatStrings("Dialog '", infoSymbName), "' not found"));
+        passed = FALSE;
+    };
+
+    // Check if dialog function exists
+    var int funcId; funcId = MEM_GetSymbolIndex(funcSymbName);
+    if (funcId == -1) {
+        G1CP_TestsuiteErrorDetail(ConcatStrings(ConcatStrings("Dialog function '", funcSymbName), "' not found"));
+        passed = FALSE;
+    };
+
+    // Check if the constant exists
+    if (MEM_GetSymbolIndex(topicSymbName) == -1) {
+        G1CP_TestsuiteErrorDetail(ConcatStrings(ConcatStrings("Variable '", topicSymbName), "' not found"));
+        passed = FALSE;
+    };
+
+    // At the latest now, we need to stop if there are fails already
+    if (!passed) {
+        return FALSE;
+    };
+
+    // Retrieve the content of the log topic string constant
+    var string topic; topic = G1CP_GetStringConst(topicSymbName, 0, "G1CP invalid string");
+
+    // First test: Check if the dialog function creates the topic if it did not exist beforehand
+
+    // Rename the log topic if it already exists
+    G1CP_LogRenameTopic(topic, tempTopicName);
+
+    // Backup and set self and other
+    var C_Npc slfBak; slfBak = MEM_CpyInst(self);
+    var C_Npc othBak; othBak = MEM_CpyInst(other);
+    self  = MEM_CpyInst(hero);
+    other = MEM_CpyInst(hero);
+
+    // Just run the dialog and see what happens
+    MEM_CallByID(funcId);
+
+    // Stop the output units
+    Npc_ClearAIQueue(hero);
+    AI_StandUpQuick(hero);
+
+    // Restore self and other
+    self  = MEM_CpyInst(slfBak);
+    other = MEM_CpyInst(othBak);
+
+    // Check if the log entry was created
+    if (!G1CP_LogGetTopic(topic)) {
+        G1CP_TestsuiteErrorDetail("Log topic was not created by the dialog function");
+        passed = FALSE;
+    };
+
+    // Clean up
+    G1CP_LogRemoveTopic(topic);
+
+    // Second test: Check if the log entry is created on applying the fix
+    // This second test is specific to how the G1CP fixes the bug and will fail if a mod fixed the issue in another way
+    var int r;
+
+    // Backup the told status of the dialog
+    var int toldBak; toldBak = Npc_KnowsInfo(hero, infoId);
+
+    // Revert the fix
+    r = G1CP_203_LogEntryGrahamMerchantRevert();
+
+    // Set the dialog to told
+    G1CP_SetInfoTold(infoSymbName, TRUE);
+
+    // Re-apply the fix
+    r = G1CP_203_LogEntryGrahamMerchant();
+
+    // Check if the topic was created
+    if (!G1CP_LogGetTopic(topic)) {
+        G1CP_TestsuiteErrorDetail("Log topic was not auto-created on applying the fix");
+        passed = FALSE;
+    };
+
+    // Revert everything
+    r = G1CP_203_LogEntryGrahamMerchantRevert();
+    G1CP_SetInfoTold(infoSymbName, toldBak);
+    r = G1CP_203_LogEntryGrahamMerchant();
+    G1CP_LogRemoveTopic(topic);
+    G1CP_LogRenameTopic(tempTopicName, topic);
```

Comment:
One thing to keep in mind with this test (in case we use it somewhere else in the future), is that it breaks the fix-lookup-table. Applying and reverting the fix like done here, does not update the fix status. In this particular function, the fix status will remain "applied" even the fix is not. The fix-lookup-table will be corrupted from that point on.
Maybe we could backup the current status of the fix in the beginning with `G1CP_GetFixStatus` and set it in the end with `G1CP_SetFixStatus`.

Reply:
I think, since this is just a test, this is not worth it. I would suggest to keep in mind that tests may just break the game.

Reply:
Since this is just a test, I think this is not so important, but feel free to fix it
```D
var int fixStatusBak; fixStatusBak = G1CP_GetFixStatus(203);

// ...

G1CP_SetFixStatus(203, fixStatusBak);
```
If you do fix this, there is some more tests where this can be done. But it's only applicable to tests where the fix-status is changed **inside the apply function** - this seems to only happen for these sorts of log entry fixes.